### PR TITLE
Add --ide flag to verify-config, changing some errors into warnings.

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: #1979 

Adds the optional `--ide` flag to `mirrord verify-config [--ide] --path {/config/path}`.

Creates a special case for `config::verify` where we may turn some errors (target related) to warnings when we're in an IDE context, as the user may select a target instead of putting on in the config file.

### CODE REVIEW

File: mirrord/config/src/lib.rs [hunk 1]
```diff
@@ -381,24 +386,67 @@ impl LayerConfig {
...
+            // In the IDE, a target may be selected after `mirrord verify-config` is run, so we
+            // for this case we treat these as warnings. They'll become errors once mirrord proper
+            // tries to start (if the user somehow managed to not select a target by then).
+            if ide {
+                if self.target.namespace.is_some() {
+                    context.add_warning(
+                        "A target namespace was specified, but no target was specified. \
+                            The IDE will prompt you to select a target."
+                            .into(),
+                    );
```
Comment (infiniteregrets) [inline mirrord/config/src/lib.rs:402]: && !self.target.path...?

File: mirrord/cli/src/config.rs [hunk 2]
```diff
@@ -62,7 +62,6 @@ pub(super) enum Commands {
     InternalProxy,
 
     /// Verify config file without starting mirrord.
-    #[command(hide = true)]
     VerifyConfig(VerifyConfigArgs),
```
Comment (infiniteregrets) [inline mirrord/cli/src/config.rs:65]: i am not sure if i am aware of the context here, do people really want to use it with the cli? why remove the hidden flag?

File: mirrord/config/src/lib.rs [hunk 3]
```diff
@@ -323,7 +323,11 @@ impl LayerConfig {
     /// We don't call it from `from_env` since we want to verify it only once (from cli)
     ///
     /// Fills `context` with the warnings.
-    pub fn verify(&self, context: &mut ConfigContext) -> Result<(), ConfigError> {
+    ///
+    /// - `ide`: Identifies if this is being called from an IDE context, when using
+    /// `mirrord verify-config`. Turns some _target missing_ errors into warnings, as the target can
+    /// be selected after `verify-config` is run.
+    pub fn verify(&self, context: &mut ConfigContext, ide: bool) -> Result<(), ConfigError> {
```
Comment (aviramha) [inline mirrord/config/src/lib.rs:330]: maybe it's better to have it under `ConfigContext`?

File: mirrord/cli/src/config.rs (hunk 2 above)
Comment (meowjesty) [reply mirrord/cli/src/config.rs:65]: Shouldn't be removed, missed this while I was testing some stuff.

File: mirrord/config/src/lib.rs (hunk 3 above)
Comment (meowjesty) [reply mirrord/config/src/lib.rs]: You meant just the `ide` arg, right?

Comment (aviramha) [reply mirrord/config/src/lib.rs]: yes

File: mirrord/config/src/lib.rs [hunk 4]
```diff
@@ -381,24 +386,70 @@ impl LayerConfig {
...
-            if self.pause {
-                Err(ConfigError::Conflict(
-                    "The target pause feature is not compatible with a targetless agent, please either disable this option or specify a target.".into(),
-                ))?;
+            // In the IDE, a target may be selected after `mirrord verify-config` is run, so we
+            // for this case we treat these as warnings. They'll become errors once mirrord proper
+            // tries to start (if the user somehow managed to not select a target by then).
+            if context.ide {
+                if self.target.namespace.is_some() {
+                    context.add_warning(
```
Comment (aviramha) [inline mirrord/config/src/lib.rs:398]: This shouldn't be a warning

**Review by infiniteregrets — APPROVED** [review]

File: mirrord/config/src/lib.rs [hunk 5]
```diff
@@ -381,24 +386,70 @@ impl LayerConfig {
...
+                        "A target namespace was specified, but no target was detected in config. \
+                        The IDE will prompt you to select a target."
+                            .into(),
+                    );
+                }
+
+                if self.feature.network.incoming.is_steal() {
```
Comment (aviramha) [inline mirrord/config/src/lib.rs:405]: Same

File: mirrord/config/src/lib.rs [hunk 6]
```diff
@@ -381,24 +386,70 @@ impl LayerConfig {
...
+                    context.add_warning(
+                        "Steal mode is not compatible with a targetless agent.\
+                        The IDE will prompt you to select a target."
+                            .into(),
+                    );
+                }
+
+                if self.agent.ephemeral {
```
Comment (aviramha) [inline mirrord/config/src/lib.rs:413]: Same

File: mirrord/config/src/lib.rs [hunk 7]
```diff
@@ -381,24 +386,70 @@ impl LayerConfig {
...
+                    context.add_warning(
+                        "Using an ephemeral container for the agent is not compatible with a \
+                        targetless agent. The IDE will prompt you to select a target."
+                            .into(),
+                    );
+                }
+
+                if self.pause {
+                    context.add_warning(
```
Comment (aviramha) [inline mirrord/config/src/lib.rs:422]: Same

**Pull request merged**